### DISCUSSION TITLE
What are your use cases? How do you use this tool?

### DISCUSSION THREAD

**Discussion by gerardbosch** [erikw/restic-automatic-backup-scheduler#53] [General]
This thread is just to listen how other people is managing their backups, what do they backup and to learn from the setup and experiences of each other.

I will start :D 

---

I use this tool just to make a backup of my homedir of my computer, so all I have is `BACKUP_PATHS=/home/myuser`. For the rest, the defaults (and default profile) works fine to me. This gives me a the first 10GB for free of storage as per Backblaze B2 conditions at the time of writing.

- OS: Linux (may extend to MacOS at some point in time, by creating a separate bucket or account)

In case of disaster, my idea is just manually reinstall my Linux, and after that, install restic and do:

```
restic restore latest --target /
```

That should restore my backup path above into `/`, overwriting the newly created `/home/myuser` with the contents of my latest snapshot.
- I try to keep most of my settings and setup as dotfiles in home instead of modifying system files, so I don't expect much problem doing OS setup after that.
- Finally, do a manual installation of this script following the README in this repo to enable automated backups again.

**Comment by erikw** [comment]
Great idea!

When I started out with restic, my goal was to have full-system remote (client side encrypted incremental cheap) backup, so that I could restore parts of the system in case of a mess-up. On this first system (Arch Linux) I had a few different backup  levels where this restic solution is one of the off-site parts. After messing around a lot and experienced many full system failures in life, I've become a bit concerned (obsessed?) with making backups :)

* Local
  * 1. COW filesystem (BTRFS or ZFS) with hourly, daily, monthly snapshots using snapper. Fast backup, to restore fast and for files that are not synced to other backups.  Using OpenSUSE's [snapper](https://wiki.archlinux.org/title/Snapper) and for BSD [zfstools](eduardosanchez.me/2015/08/31/zfs-automatic-snapshots/)
* Off-system
  * Mid-term backup on USB disk: using rsync to emulate Apple's Time Machine:  [time_machine.sh](https://github.com/erikw/dotfiles/blob/personal/bin/archived/time_machine.sh). Started when plugging in the disk, or as cronjob as long as plugged in.
* Off-site
  * Restic on scheduled timer (this project). Backups not the whole system, I ignore system files and binaries and aim to include files that I crafted myself like in `/etc`. I work with a blocklist instead of an includelist though, as I rather include too much than too little in my backup.
  * Git: all source code-like projects that can live in a git repo are in public or private GitHub repositories. Using the excellent tool [ghq](https://github.com/x-motemen/ghq) to mirror the git clones structure from the remote end to my local disk. Using this tool I will clone e.g. this repo directly to `~/src/github.com/erikw/restic-systemd-automatic-backup`



With this setup, the most important documents are backed up at many places and the likelihood of me losing an important file is small!

Since the first system, I also used `restic_backup.sh` in Windows, macOS and freeBSD in the same manner as cronjob or Windows Scheduled tasks.

**Reply by gerardbosch** [reply]
A really backup obsession indeed :smile:
